velox/functions/prestosql/aggregates/MinMaxAggregates.cpp

```diff
@@ -180,6 +180,31 @@ class MinAggregate : public MinMaxAggregate<T> {
     }
   }
 
+  bool supportsToIntermediate() const override {
+    return true;
+  }
+
+  void toIntermediate(
+      const SelectivityVector& rows,
+      std::vector<VectorPtr>& args,
+      VectorPtr& result) const override {
+    const auto& input = args[0];
+    if (rows.isAllSelected()) {
+      result = input;
+      return;
+    }
+
+    auto* pool = BaseAggregate::allocator_->pool();
+
+    result = BaseVector::create(input->type(), rows.size(), pool);
+    result->copy(input.get(), 0, 0, rows.size());
+
+    // Set result to NULL for rows that are masked out.
+    BufferPtr nulls = allocateNulls(rows.size(), pool, bits::kNull);
+    rows.clearNulls(nulls);
```

Comment:
maybe I'm too picky (and not related to this PR), but having this API in selectivity vector and not on the null buffer itself is a bit counter-intuitive. This might be more intuitive:

```
result->setNulls(allocateNulls(rows.size(), pool, bits::kNull).clearNulls(rows));
```

or even better, maybe allocateNulls could take the selectivity vector directly?

```
result->setNulls(allocateNulls(rows, pool));
```

Reply:
This is an interesting idea. Nulls buffer is just a byte buffer that doesn't know that it contains nulls and there is currently no dependency from buffers into vectors.

Reply:
Makes sense. I guess we would specialize Buffer into sort of a NullBuffer in that case to add these null specific routines, but no big deal.